clustered/client/src/main/java/org/ehcache/clustered/client/internal/ConnectionSource.java

```diff
@@ -81,22 +90,58 @@ private static URI extractClusterUri(URI uri) {
 
   public static class ServerList extends ConnectionSource {
 
-    private final Iterable<InetSocketAddress> servers;
+    private final CopyOnWriteArraySet<InetSocketAddress> servers;
     private final String clusterTierManager;
 
     public ServerList(Iterable<InetSocketAddress> servers, String clusterTierManager) {
-      this.servers = cloneServers(Objects.requireNonNull(servers, "Servers cannot be null"));
+      this.servers = createServerList(servers);
       this.clusterTierManager = Objects.requireNonNull(clusterTierManager, "Cluster tier manager identifier cannot be null");
     }
 
+    private CopyOnWriteArraySet<InetSocketAddress> createServerList(Iterable<InetSocketAddress> servers) {
+      Objects.requireNonNull(servers, "Servers cannot be null");
+      CopyOnWriteArraySet<InetSocketAddress> serverList = new CopyOnWriteArraySet<>();
+      servers.forEach(server -> serverList.add(server));
+      return serverList;
+    }
+
     @Override
     public String getClusterTierManager() {
       return clusterTierManager;
     }
 
     @Override
     public LeasedConnection connect(Properties connectionProperties) throws ConnectionException {
-      return LeasedConnectionFactory.connect(servers, connectionProperties);
+      LeasedConnection connection = LeasedConnectionFactory.connect(servers, connectionProperties);
+      try {
+        EntityRef<DynamicTopologyEntity, Object, DynamicTopologyEntity.Settings> ref = connection.getEntityRef(DynamicTopologyEntity.class, 1, DynamicTopologyEntityConstants.ENTITY_NAME);
```

Comment:
Where does this version number come from and how do we make sure it stays in sync with the server side?

Reply:
We can introduce some constants for version number as well but currently it doesn't have that.